filter.go

```diff
@@ -0,0 +1,90 @@
+package inflight
+
+import (
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+)
+
+type FQFilter struct {
+	// list-watching API models
+	lock   *sync.Mutex
+	queues []*Queue
+
+	*sharedDispatcher
+
+	Matcher  func(*http.Request, []*Queue) *Queue
+	Delegate http.HandlerFunc
+}
+
+func (f *FQFilter) Serve(resp http.ResponseWriter, req *http.Request) {
+
+	matchedQueue := f.Matcher(req, f.queues)
+	// fmt.Printf("matchedQueue: %d\n", matchedQueue.Index)
+
+	dispatcher := f.sharedDispatcher.producers[matchedQueue.Priority]
+	func() {
+		dispatcher.lock.Lock()
+		defer dispatcher.lock.Unlock()
+		// dispatcher.queues[0].Priority
+
+		// fmt.Printf("dispatcher.requestsexecuting: %d\n", dispatcher.requestsexecuting)
+		if dispatcher.GetRequestsExecuting() > dispatcher.ACV {
+			// too many requests
+			// fmt.Println("throttled...")
+			resp.WriteHeader(http.StatusConflict)
+			return
+		}
+	}()
+	// TODO(aaron-prindle) currently the line below is a bit hacky..
```

Comment:
I see that no locks are held at this point.  Any decisions made above could be invalidated at this point.